src/test/java/com/artipie/pypi/PypiITCase.java

```diff
@@ -23,205 +23,85 @@
+import org.testcontainers.containers.BindMode;
 
 /**
  * Integration tests for Pypi repository.
+ *
  * @checkstyle ClassDataAbstractionCouplingCheck (500 lines)
  * @since 0.12
  */
 @SuppressWarnings("PMD.AvoidDuplicateLiterals")
 @EnabledOnOs({OS.LINUX, OS.MAC})
 final class PypiITCase {
-
-    /**
-     * Host.
-     */
-    private static final String HOST = "host.testcontainers.internal";
-
     /**
-     * Temporary directory for all tests.
-     * @checkstyle VisibilityModifierCheck (3 lines)
+     * Test deployments.
+     *
+     * @checkstyle VisibilityModifierCheck (10 lines)
      */
-    @TempDir
-    Path tmp;
-
-    /**
-     * Tested Artipie server.
-     */
-    private ArtipieServer server;
-
-    /**
-     * Container.
-     */
-    private TestContainer cntn;
-
-    /**
-     * Storage.
-     */
-    private Storage storage;
-
-    /**
-     * Repository url.
-     */
-    private RepositoryUrl url;
-
-    @ParameterizedTest
-    @ValueSource(booleans = {true, false})
-    void pypiInstall(final boolean anonymous) throws Exception {
-        this.init(anonymous);
-        new TestResource("pypi-repo/alarmtime-0.1.5.tar.gz")
-            .saveTo(
-                this.storage,
-                new Key.From("repos", "my-pypi", "alarmtime", "alarmtime-0.1.5.tar.gz")
-        );
-        MatcherAssert.assertThat(
-            this.cntn.execStdout(
-                "pip", "install", "--no-deps", "--trusted-host", PypiITCase.HOST,
-                "--index-url", this.url.string(anonymous),
-                "alarmtime==0.1.5"
-            ),
-            new StringContains("Successfully installed alarmtime-0.1.5")
-        );
-    }
-
-    @ParameterizedTest
-    @ValueSource(booleans = {true, false})
-    void pypiPublish(final boolean anonymous) throws Exception {
-        this.init(anonymous);
-        new TestResource("pypi-repo/example-pckg/dist/")
-            .addFilesTo(
-                this.storage,
-                new Key.From("pypi-repo", "example-pckg")
-        );
-        this.cntn.execStdout("python3", "-m", "pip", "install", "--user", "--upgrade", "twine");
-        MatcherAssert.assertThat(
-            "Packages should be uploaded",
-            this.cntn.execStdout(
-                "python3", "-m", "twine", "upload", "--repository-url", this.url.string(),
-                "-u", ArtipieServer.ALICE.name(), "-p", this.pswd(anonymous),
-                "pypi-repo/example-pckg/*"
-            ),
-            new StringContainsInOrder(
-                new ListOf<String>(
-                    "Uploading artipietestpkg-0.0.3-py2-none-any.whl", "100%",
-                    "Uploading artipietestpkg-0.0.3.tar.gz", "100%"
-                )
+    @RegisterExtension
+    final TestDeployment containers = new TestDeployment(
+        () -> TestDeployment.ArtipieContainer.defaultDefinition()
+            .withRepoConfig("pypi-repo/pypi.yml", "my-python"),
+        () -> new TestDeployment.ClientContainer("python")
```

Comment:
@milkysoft I'd suggest to use [docker image](https://hub.docker.com/_/python?tab=tags&page=1&ordering=last_updated) with specified python version to avoid any potential problems with new python release

Reply:
@olenagerasimova , fixed with latest commit